src/DependencyManagement/DependencyManager.cs

```diff
@@ -0,0 +1,372 @@
+                DependencyManagementUtils.EmptyDirectory(DependenciesPath);
+            }
+            else
+            {
+                // If the destination path does not exist, create it.
+                Directory.CreateDirectory(DependenciesPath);
+            }
+
+            try
+            {
+                foreach (var module in Dependencies)
+                {
+                    var moduleName = module.Name;
+                    var latestVersion = module.LatestVersion;
+
+                    // Save the module to the given path
+                    pwsh.AddCommand("PowerShellGet\\Save-Module")
+                        .AddParameter("Repository", Repository)
+                        .AddParameter("Name", moduleName)
+                        .AddParameter("RequiredVersion", latestVersion)
+                        .AddParameter("Path", DependenciesPath)
+                        .AddParameter("Force", true)
+                        .AddParameter("ErrorAction", "Stop")
+                        .InvokeAndClearCommands();
+
+                    var message = string.Format(PowerShellWorkerStrings.ModuleHasBeenInstalled, moduleName, latestVersion);
+                    logger.Log(LogLevel.Trace, message, isUserLog: true);
+                }
+            }
+            finally
+            {
+                // Clean up
+                pwsh.AddCommand("Microsoft.PowerShell.Core\\Remove-Module")
+                    .AddParameter("Name", "PackageManagement, PowerShellGet")
+                    .AddParameter("Force", true)
+                    .AddParameter("ErrorAction", "SilentlyContinue")
+                    .InvokeAndClearCommands();
+            }
+        }
+
+        /// <summary>
+        /// Validates that the given major version is less or equal to the latest supported major version.
+        /// </summary>
+        private void ValidateModuleMajorVersion(string moduleName, string majorVersion, string latestVersion)
+        {
+            // A Version object cannot be created with a single digit so add a '.0' to it.
+            var requestedVersion = new Version(majorVersion + ".0");
+            var latestSupportedVersion = new Version(latestVersion);
+
+            if (requestedVersion.Major > latestSupportedVersion.Major)
+            {
+                // The requested major version is greater than the latest major supported version.
+                var errorMsg = string.Format(PowerShellWorkerStrings.InvalidModuleMajorVersion, moduleName, majorVersion);
+                var argException = new ArgumentException(errorMsg);
+                throw argException;
+            }
+        }
+
+        /// <summary>
+        /// Parses the given string version and extracts the major version.
+        /// Please note that the only version we currently support is of the form '1.*'.
+        /// </summary>
+        private string GetMajorVersion(string version)
```

Comment:
Could this just be:

```csharp
Version.Parse(version).Major
```

and then you pass around an int instead of a string.

it will come in handy on [line 214](github.com/Azure/azure-functions-powershell-worker/pull/166/files#diff-41c954a297870a25469fd5631c1e0683R214) because you can do:

```csharp
var requestedVersion = new Version(major: majorVersion, minor: 0);
```

ints are nicer to deal with than strings when it comes to versions IMO.

Reply:
That is true. However, I believe the future plan is to support more than a major version, in which using a string will come handy. Also, we need to validate that version is not null or empty and ensure it is in the correct format. Otherwise, we need to throw the appropriate exceptions, and that's what this method does. I will keep it as is.

Reply:
Makes sense!